frontend/src/features/admin-form/responses/ResponsesPage/SecretKeyVerification.tsx

```diff
@@ -0,0 +1,196 @@
+import { useCallback, useMemo, useRef } from 'react'
+import { RegisterOptions, useForm } from 'react-hook-form'
+import { BiUpload } from 'react-icons/bi'
+import {
+  Container,
+  FormControl,
+  FormErrorMessage,
+  IconButton,
+  Input,
+  Skeleton,
+  Stack,
+  Text,
+} from '@chakra-ui/react'
+import simplur from 'simplur'
+
+import { useIsMobile } from '~hooks/useIsMobile'
+import formsgSdk from '~utils/formSdk'
+import Button from '~components/Button'
+import FormLabel from '~components/FormControl/FormLabel'
+import Link from '~components/Link'
+
+import { FormActivationSvg } from '~features/admin-form/settings/components/FormActivationSvg'
+
+import { useResponsesContext } from './ResponsesContext'
+
+const SECRET_KEY_NAME = 'secretKey'
+const SECRET_KEY_REGEX = /^[a-zA-Z0-9/+]+={0,2}$/
+
+interface SecretKeyFormInputs {
+  [SECRET_KEY_NAME]: string
+}
+
+const useSecretKeyVerification = () => {
```

Comment:
nit: can move this into a different file

Reply:
ah was trying out a new pattern here to abstract away the view and the logic. think it makes more sense to keep it colocated in the same file